### PR TITLE
MCOL-5779: use encoding for correct check of ALTER TABLE CHANGE COLUMN

### PR DESCRIPTION
**Pull request by mariadb-LeonidFedorov**

### CODE REVIEW

**Review by drrtuy — APPROVED** [review]
untested ACK

File: dbcon/ddlpackageproc/altertableprocessor.cpp [hunk 1]
```diff
@@ -88,6 +90,14 @@ struct extentInfo
 
 namespace
 {
+
+bool checkTextTypesLengthAreUqual(const CalpontSystemCatalog::ColType& colType, const ColumnType& newType)
+{
+  auto* newTypeCharset = get_charset(newType.fCharsetNum, MYF(MY_WME));
+  auto newTypeBytesInChar = newTypeCharset ? newTypeCharset->mbmaxlen : 1;
+  return colType.colWidth * newTypeBytesInChar == newType.fLength * colType.getCharset()->mbmaxlen;
```
Comment (mariadb-SergeyZefirov) [inline dbcon/ddlpackageproc/altertableprocessor.cpp]: If the intention to check for equality of max text len of old and new types, then why do we multiply old width by new mbMaxLen and new width by (apparently) old mbMaxLen?

This need a comment on why is that.

**Review by mariadb-SergeyZefirov — APPROVED** [review]
I generally agree with what is done, but crucial piece of code needs a comment. ;)

**Review by mariadb-SergeyZefirov — APPROVED** [review]
I completely agree.

**Pull request merged**